PWGCF/TwoParticleCorrelations/Tasks/lambdaR2Correlation.cxx

```diff
@@ -336,6 +385,66 @@ struct lambdaCorrTableProducer {
     }
   }
 
+  template <ParticleType v0part, PidType pos_prong, PidType neg_prong, MassWindowType masswin, typename C, typename V, typename T>
+  void selV0Particle(C const& collision, V const& v0track, T const& tracks)
+  {
+
+    // initialize variables
+    auto postrack = v0track.template posTrack_as<T>();
+    auto negtrack = v0track.template negTrack_as<T>();
+    float mass = 0., rap = 0.;
+
+    if (v0part == kLambda) {
+      mass = v0track.mLambda();
+    }
+    if (v0part == kAntiLambda) {
+      mass = v0track.mAntiLambda();
+    }
+
+    if (cfg_do_eta_analysis) {
+      rap = v0track.eta();
+    } else {
+      rap = v0track.yLambda();
+    }
+
+    // apply mass window selection [global]
+    if ((fabs(mass - MassLambda0) >= cfg_lambda_mass_window) || (fabs(v0track.mK0Short() - MassK0Short) <= cfg_kshort_rej)) {
+      return;
+    }
+
+    // apply daughter particle id
+    if (!selPIDTrack<pos_prong>(postrack) || !selPIDTrack<neg_prong>(negtrack)) {
+      return;
+    }
```

Comment:
This will allow to say if it is a Lambda, antiLambda, or something else, right? So it should be first